packages/lambdas/src/sqs-to-converter.ts

```diff
@@ -59,24 +65,123 @@ function isSidechainConnector() {
   return sidechainFHIRConverterUrl ? true : false;
 }
 
-async function getSidechainConverterAPIKey(): Promise<string> {
-  if (!sidechainFHIRConverterKeysSecretName) {
-    throw new Error(`Programming error - sidechainFHIRConverterKeys is not set`);
+async function getAndUpdateSidechainConverterKeys(): Promise<string> {
+  if (!sidechainKeysTableName) {
+    throw new Error(`Programming error - SIDECHAIN_FHIR_CONVERTER_KEYS_TABLE_NAME is not set`);
   }
 
-  const secret = await getSecret(sidechainFHIRConverterKeysSecretName);
-  if (!secret) {
-    throw new Error(`Config error - sidechainFHIRConverterKeysSecret doesn't exist`);
+  const docClient = new DynamoDB.DocumentClient({
+    apiVersion: "2012-08-10",
+  });
+
+  // get all keys that haven't been revoked
+  const keysTableItems = await docClient
+    .scan({
+      TableName: sidechainKeysTableName,
+      FilterExpression: "attribute_exists(status) AND (status = :active OR status = :rateLimit)",
+      ExpressionAttributeValues: {
+        ":active": sidechainKeysStatus.active.toString(),
+        ":rateLimit": sidechainKeysStatus.rateLimit.toString(),
+      },
+    })
+    .promise();
+
+  if (!keysTableItems.Items) {
+    throw new Error(`Programming error - no keys found in table ${sidechainKeysTableName}`);
+  }
+  let activeKey = "";
+  const keysToUpdate: string[] = [];
+  for (const keyItem of keysTableItems.Items) {
+    if (!keyItem.status || !keyItem.key) {
+      throw new Error(
+        `Programming error - keys found in table ${sidechainKeysTableName} are not the expected format`
+      );
+    }
+    const rateLimitDate = keyItem.rateLimitDate ? dayjs(keyItem.rateLimitDate) : undefined;
+    const now = dayjs();
+    if (keyItem.status === sidechainKeysStatus.active.toString()) {
+      if (!activeKey) {
+        // pick the first active API key
+        activeKey = keyItem.key;
+      }
+    } else if (
+      keyItem.status === sidechainKeysStatus.rateLimit.toString() &&
+      rateLimitDate &&
```

Comment:
I was going to suggest error'ing if `rateLimitDate` was not present in this case, but I think we should just remove the rate limit status.

The presence of the rate limit date and whether its on the current month is the actual indicative of this state, having a status for this as well is redundant.

Then we would have only 2 status, active or revoked.

Reply:
IMO having an explicit status is better than making it implicit based on whether a date is present - it's only obvious to someone that wrote the code